web/wp-content/plugins/tmbi-bx/inc/bx-base.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * Created by PhpStorm.
+ * User: MJ
+ * Date: 3/28/2017
+ * Time: 1:10 PM
+ */
+class BX_Base {
+	const FILE_SPEC         = __DIR__;
+	const LOCALIZED_LABEL   = 'rd_bx';
+	const LOADER_SCRIPT     = 'js/smart-tag.js';
+	const LOADER_LABEL      = 'fhm-bx-loader';
+	const SCRIPT_ID         = 'default';
+	const DEPENDS           = '';
+	static public $version;
+
+	public function set_version( $version ) {
+		static::$version = $version;
+	}
+
+	public function render_bx_load_failure() {
+		print( '<!-- bounce exchange script has failed to load -->' . PHP_EOL );
+	}
+
+	public function render_bx_is_blocked() {
+		print( '<!-- Bounce Exchange has been blocked -->' . PHP_EOL );
+	}
+
+	public function init() {
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	public function enqueue_scripts() {
+		wp_register_script(
+			static::LOADER_LABEL,
+			self::get_asset_url( static::LOADER_SCRIPT ),
+			static::DEPENDS,
+			static::$version,
```

Comment:
What's the point of using a public static variable, that is filled from a setter (`set_version`), that's only called with a constant param (`$this->bx->set_version( self::VERSION );`)? That's a lot of code running just to pass the script version to the tag.